magefile.go

```diff
@@ -3829,9 +3872,147 @@ func updateYamlFile(path string, keyVal ...struct {
 	return nil
 }
 
+func (Helm) ensureRepository(repoName, repoURL string, settings *cli.EnvSettings) error {
+	repoFile := settings.RepositoryConfig
+	// Load existing repositories
+	file, err := repo.LoadFile(repoFile)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			file = repo.NewFile()
+		} else {
+			return fmt.Errorf("could not load Helm repository config: %w", err)
+		}
+	}
+
+	// Check if the repository is already added
+	for _, entry := range file.Repositories {
+		if entry.URL == repoURL {
+			// repository already exists
+			return nil
+		}
+	}
+
+	// Add the repository
+	entry := &repo.Entry{
+		Name: repoName,
+		URL:  repoURL,
+	}
+
+	chartRepo, err := repo.NewChartRepository(entry, getter.All(settings))
+	if err != nil {
+		return fmt.Errorf("could not create repo %s: %w", repoURL, err)
+	}
+
+	_, err = chartRepo.DownloadIndexFile()
+	if err != nil {
+		return fmt.Errorf("could not download index file for repo %s: %w", repoURL, err)
+	}
+
+	file.Update(entry)
+	if err := file.WriteFile(repoFile, 0o644); err != nil {
+		return fmt.Errorf("could not write Helm repository config: %w", err)
+	}
+
+	return nil
+}
+
 // BuildDependencies builds the dependencies for the Elastic-Agent Helm chart.
-func (Helm) BuildDependencies() error {
-	return helm.BuildChartDependencies(helmChartPath)
+func (h Helm) BuildDependencies() error {
```

Comment:
How is the implementation here different from calling `helm dependency update`? Worth adding a comment explaining why we're doing something custom.